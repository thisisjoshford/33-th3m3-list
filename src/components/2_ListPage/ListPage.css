.Dark{
  background-color: black;
  color: white;
}

.Light{
  background-color: white;
  color: blue;
}