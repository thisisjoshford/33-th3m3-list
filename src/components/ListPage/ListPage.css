.Dark{
  background-color: black;
  color: white;
  html {
    background-color: black;
  }
}

.Light{
  background-color: white;
  color: blue;
  html {
    background-color: white;
  }
}