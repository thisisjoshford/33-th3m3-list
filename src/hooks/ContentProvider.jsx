import React, { createContext, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { getPeople } from '../services/apiFetch';

export const XfilesContent = createContext();

export const ContentProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);

  useEffect(() => {
    getPeople()
      .then(peeps => setCharacters(peeps));
  }, []);

  return (
    <XfilesContent.Provider value={characters}>
      {children}
    </XfilesContent.Provider>
  );
};

ContentProvider.propTypes = {
  children: PropTypes.node
};

