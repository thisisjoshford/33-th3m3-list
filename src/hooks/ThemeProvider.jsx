import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

const ThemeContext = createContext();

export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');
  const toggle = ({ target }) => {
    if(target.checked) setTheme('dark');
    if(!target.checked) setTheme('light');
  };

  return (
    <ThemeContext.Provider value={{ theme, toggle }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.node
};

export const useTheme = () => {
  const theme = useContext(ThemeContext);
  return theme;
};

export const useToggle = () => {
  const toggle = useContext(ThemeContext);
  return toggle;
};
