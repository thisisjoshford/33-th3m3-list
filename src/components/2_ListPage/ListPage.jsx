import React, { useContext } from 'react';
import { XfilesContent } from '../../hooks/ContentProvider';
import { useTheme } from '../../hooks/ThemeProvider';

const ListPage = () => {
  const { characters } = useContext(XfilesContent);
  const { theme } = useTheme();
  
  let listHeader;
  if(theme === 'light') {listHeader = (<h3> LIGHT!!!!!! </h3>);}
  if(theme === 'dark') {listHeader = (<h3> DARK!!!!!!</h3>);}
  
  const elements = characters.map((character) => (
    <li key={character.name}>{character.name}</li>
  ));

  return (
    <>
      {listHeader}
      <ul>
        { elements }  
      </ul>
    </>
  );
};

export default ListPage;


// const elements = characters.map((character) => (
//   <div key={character.id}> 
//     <img src={character.image} />
//     <h3>{character.name}</h3>
//     <h6>{character.occupation}</h6>
//     <p>{character.description}</p>
//   </div>
// ));