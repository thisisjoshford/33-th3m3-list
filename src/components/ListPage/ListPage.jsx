import React, { useContext } from 'react';
import { XfilesContent } from '../../hooks/ContentProvider';
import { useTheme } from '../../hooks/ThemeProvider';
import ListStyle from './ListPage.css';

const ListPage = () => {
  const { characters, page, paging } = useContext(XfilesContent);
  const { theme } = useTheme();
  
  let styling;
  if(theme === 'light') styling = ListStyle.Light;
  if(theme === 'dark') styling = ListStyle.Dark;
  
  const elements = characters.map((character) => (
    <li key={character.name}>{character.name}</li>
  ));

  return (
    <>
      <br></br>

      <button 
        onClick={() => paging(-1)} 
        disabled={page === 1}
      >&#8672;
      </button>

      <button 
        onClick={() => paging(1)}
        disabled={characters.length < 33}
      >&#8674;
      </button>

      <ul className={styling}>  
        { elements }  
      </ul>

    </>
  );
};

export default ListPage;


// const elements = characters.map((character) => (
//   <div key={character.id}> 
//     <img src={character.image} />
//     <h3>{character.name}</h3>
//     <h6>{character.occupation}</h6>
//     <p>{character.description}</p>
//   </div>
// ));