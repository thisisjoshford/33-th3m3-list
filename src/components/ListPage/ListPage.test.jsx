import React from 'react';
import { shallow } from 'enzyme';
import ListPage from '../ListPage/ListPage';
import { ThemeProvider } from '../../hooks/ThemeProvider';

describe('ListPage component', () => {

  it('renders ListPage', () => {
    const wrapper = shallow(
      <ThemeProvider>
        <ListPage/>
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
