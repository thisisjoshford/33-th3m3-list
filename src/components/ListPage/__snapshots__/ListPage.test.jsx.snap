// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage component renders ListPage 1`] = `
<ContextProvider
  value={
    Object {
      "theme": "light",
      "toggle": [Function],
    }
  }
>
  <ListPage />
</ContextProvider>
`;
